#Novel strain discovery, based on read depth (Matt)  
##OTUs (Operational Taxonomic Units)    
- Rarefaction Curve, estimation of complexity (10.1371/journal.pcbi.1000667)
- Assume no bias in sampling (PCR, over representation of certain sequences), no technical errors  
- These curves only work when you don't have rare species in the samples, or when the **beta diversity** is high  
	- No overlap between groups, similarity within the group pretty high (assumedly)  

##Making a good study model for metagenomics
- If we know the complexity of our samples, we can figure out how many human gut samples we need for microbiome analysis  
- An older model (Lander-Waterman statistic), assumes everything is perfect, used to estimate coverage for a single species   
 
#OTUs
- A more precise definition or basis for grouping.  

#Novel gene discovery  
- Most of it comes from where the study is specifically modelled to identify these novel genes.  
- M5nr <- non redundant db with protein sequences and annotaitons from multiple sources and associated tools.  
- In vitro culture is a big hurdle to cross  

#Markget gene based approach
- Genes that are not prone to HGT, genes that are single copy  
	- How do they identify these?  



